From 0b6c9d93e38b891033cb7bbca140b8074b77bb23 Mon Sep 17 00:00:00 2001
From: Robbie Harwood <rharwood@redhat.com>
Date: Wed, 13 Jan 2016 18:17:09 -0500
Subject: [PATCH] Don't ignore kadm5.acl without trailing \n on non-x86

On platforms where the char type is not unsigned, the check for EOF will
always fail.  This causes the contents of kadm5.acl to be ignored.
---
 src/lib/kadm5/srv/server_acl.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/lib/kadm5/srv/server_acl.c b/src/lib/kadm5/srv/server_acl.c
index bfeccfd..72ac524 100644
--- a/src/lib/kadm5/srv/server_acl.c
+++ b/src/lib/kadm5/srv/server_acl.c
@@ -115,7 +115,9 @@ kadm5int_acl_get_line(fp, lnp)
             int byte;
             byte = fgetc(fp);
             acl_buf[i] = byte;
-            if (byte == (char)EOF) {
+
+            /* fgetc()'s return value, EOF, and `byte` are all ints. */
+            if (byte == EOF) {
                 if (i > 0 && acl_buf[i-1] == '\\')
                     i--;
                 break;          /* it gets nulled-out below */
-- 
2.6.4

